kloppy/domain/models/common.py

```diff
@@ -940,28 +941,92 @@ def to_records(
         else:
             return iterator
 
+    def to_dict(
+        self,
+        *columns: "Column",
+        **named_columns: "Column",
+    ) -> Dict[str, List[Any]]:
+
+        from ..services.transformers.data_record import get_transformer_cls
+
+        transformer = get_transformer_cls(self.dataset_type)(
+            *columns, **named_columns
+        )
+
+        c = len(self.records)
+        items = defaultdict(lambda: [None] * c)
+        for i, record in enumerate(self.records):
+            item = transformer(record)
+            for k, v in item.items():
+                items[k][i] = v
+
+        return items
+
     def to_df(
         self,
         *columns: "Column",
-        engine: Optional[Union[Literal["polars"], Literal["pandas"]]] = None,
+        engine: Optional[
+            Union[
+                Literal["polars"],
+                Literal["pandas"],
+                Literal["pandas[pyarrow]"],
+            ]
+        ] = None,
         **named_columns: "Column",
     ):
         from kloppy.config import get_config
 
         if not engine:
             engine = get_config("dataframe.engine")
 
-        if engine == "pandas":
-            from pandas import DataFrame
+        if engine == "pandas[pyarrow]":
+            try:
```

Comment:
Do we need this `try-except` construct when the `engine` is set to `pandas[pyarrow]`? Wouldn't it be enough to check whether `ArrowDtype` can be imported from `pandas`?